src/Components/Trainers/Form/index.jsx

```diff
@@ -109,96 +95,97 @@ const Form = () => {
-            placeholder="Phone Number"
-            value={trainer.phone}
-            onChange={(e) => setTrainer({ ...trainer, phone: e.target.value })}
-            required
+          <Input
+            register={register}
+            labelName={'Phone Number'}
+            inputType={'number'}
+            inputName={'phone'}
+            error={errors.phone?.message}
           />
         </div>
         <div className={styles.box}>
-          <h4>Email</h4>
-          <input
-            type="text"
-            placeholder="Email"
-            value={trainer.email}
-            onChange={(e) => setTrainer({ ...trainer, email: e.target.value })}
-            required
+          <Input
+            register={register}
+            labelName={'Email'}
+            inputType={'text'}
+            inputName={'email'}
+            error={errors.email?.message}
           />
         </div>
         <div className={styles.box}>
-          <h4>City</h4>
-          <input
-            type="text"
-            placeholder="City"
-            value={trainer.city}
-            onChange={(e) => setTrainer({ ...trainer, city: e.target.value })}
-            required
+          <Input
+            register={register}
+            labelName={'City'}
+            inputType={'text'}
+            inputName={'city'}
+            error={errors.city?.message}
           />
         </div>
         <div className={styles.box}>
-          <h4>Salary</h4>
-          <input
-            type="number"
-            placeholder="Salary"
-            value={trainer.salary}
-            onChange={(e) => setTrainer({ ...trainer, salary: e.target.value })}
-            required
+          <Input
+            register={register}
+            labelName={'Salary'}
+            inputType={'number'}
+            inputName={'salary'}
+            error={errors.salary?.message}
           />
         </div>
         <div className={styles.box}>
-          <h4>Password</h4>
-          <input
-            type="text"
-            placeholder="Password"
-            value={trainer.password}
-            onChange={(e) => setTrainer({ ...trainer, password: e.target.value })}
-            required
+          <Input
+            register={register}
+            labelName={'Password'}
+            inputType={'text'}
+            inputName={'password'}
+            error={errors.password?.message}
           />
         </div>
       </div>
-      <div className={styles.buttons}>
-        <Button text="Cancel" type="cancel" clickAction={handleCancel} />
-        <Button text={id ? 'Confirm' : 'Submit'} type="submit" clickAction={handleSubmit} />
+      <div>
+        <div className={styles.buttons}>
+          <Button text="Back" type="cancel" clickAction={handleCancel} />
+          <Button text={id ? 'Update' : 'Add'} type="submit" />
+        </div>
+        <div className={styles.buttons}>
+          <Button type={'cancel'} clickAction={handleReset} text={'Reset'} />
```

Comment:
Your reset button its not working properly, when i click it does the post or put method also.